models/transmission_status.go

```diff
@@ -39,13 +40,19 @@ func (as *TransmissionStatus) UnmarshalJSON(data []byte) error {
 		return fmt.Errorf("TransmissionStatus should be a string, got %s", data)
 	}
 
-	got, err := map[string]TransmissionStatus{"FAILED": Failed, "SENT": Sent, "ACKNOWLEDGED": Acknowledged, "TRXESCALATED": Trxescalated}[s]
-	if !err {
-		return fmt.Errorf("invalid TransmissionStatus %q", s)
-	}
-	*as = got
+	new := TransmissionStatus(strings.ToUpper(s))
+	*as = new
 	return nil
 }
+
+func (as TransmissionStatus) Validate() (bool, error) {
+	_, found := map[string]TransmissionStatus{"FAILED": Failed, "SENT": Sent, "ACKNOWLEDGED": Acknowledged, "TRXESCALATED": Trxescalated}[string(as)]
+	if !found {
+		return false, NewErrContractInvalid(fmt.Sprintf("invalid Transmission Status %q", as))
+	}
+	return true, nil
+}
```

Comment:
Needs unit test

Reply:
Also transmission.go::UnmarshalJSON() needs to be covered by unit test. That's where the large drop in coverage reported by codecov.io is coming from.